imagegw/shifter_imagegw/dockerv2.py

```diff
@@ -320,20 +324,27 @@ def saveLayer(self, layer, cachedir='./'):
                 return False
         maxlen = int(r1.getheader('content-length'))
         nread = 0
-        output = open(filename, "w")
-        readsz = 4 * 1024 * 1024 # read 4MB chunks
-        while nread < maxlen:
-            buff = r1.read(readsz)
-            if buff is None:
-                break
-            if type(buff) != str:
-                print buff
+        (output_fd, output_fname) = tempfile.mkstemp('.partial', layer, cachedir)
+        output_fp = os.fdopen(output_fd, 'w')
 
-            output.write(buff)
-            nread += len(buff)
-        output.close()
-        self.checkLayerChecksum(layer, filename)
+        try:
+            readsz = 4 * 1024 * 1024 # read 4MB chunks
+            while nread < maxlen:
+                buff = r1.read(readsz)
+                if buff is None:
+                    break
+                if type(buff) != str:
```

Comment:
from offline discussion, this was about a debug line that crept in.  feel free to get rid of the debug print line